### PR TITLE
 Updates the plugin to make it work with videojs 7.

### PR DESCRIPTION
**Pull request by OliDM**
Additional changes: 
 - Adds an example page.
- Adds a parameter to disable or enable the socials filtering due to the mobile device check, this is enabled by default to avoid breaking changes.

### CODE REVIEW

**Comment by OliDM** [comment]
@avdeev after reading through the documentation, I found that since the plugin is an `advanced plugin` it uses a factory approach which is only compatible from version 6 onwards as mentioned on https://blog.videojs.com/feature-spotlight-advanced-plugins/

I have bumped up the videojs version on the package.json and travis.yml files also wrote this caveat on the readme of the project.

File: package.json [hunk 1]
```diff
@@ -74,9 +74,9 @@
     }
   },
   "dependencies": {
-    "clipboard": "^1.6.1",
-    "vanilla-sharing": "^5.1.0",
-    "video.js": "^5.19.2"
+    "clipboard": "^1.7.1",
+    "vanilla-sharing": "^5.3.2",
+    "video.js": "^7.1.0"
```
Comment (avdeev) [inline package.json]: See example of dependencies section from this generator https://github.com/videojs/generator-videojs-plugin

```
  "dependencies": {
    "global": "^4.3.2",
    "video.js": "^6 || ^7"
  },
```

**Comment by OliDM** [comment]
@avdeev nice catch, I updated the branch the dependency changes.

**Comment by avdeev** [comment]
@OliDM LGTM, thank you!
Test it, plz
I'll do release in next 2 days

**Comment by OliDM** [comment]
@avdeev All the previous tests are running against version 6 and 7 of videojs

screenshot 2018-08-21 13 02 40

is there anything extra you would like me to do?

**Comment by avdeev** [comment]
@OliDM No, thank you :)

**Comment by avdeev** [comment]
@OliDM I released in to NPM

**Pull request merged**